### PR TITLE
add support for NSTs

### PR DESCRIPTION
**Pull request by johannbarbie**
partially resolves https://github.com/leapdao/bridge-ui/issues/214

# list of tokens:
Screen Shot 2019-06-28 at 12 22 07 PM

# ability to select tokens:
Screen Shot 2019-06-28 at 12 22 47 PM

# transfer:
Screen Shot 2019-06-28 at 12 52 32 PM

### CODE REVIEW

File: src/stores/tokens.ts [hunk 1]
```diff
@@ -83,9 +85,15 @@ export class TokensStore {
       return -1;
     }
 
-    return isNFT(color)
-      ? this.erc20TokenCount + (color - NFT_COLOR_BASE)
-      : color;
+    let index = color;
```
Comment (troggy) [inline src/stores/tokens.ts]: Maybe without local vars like this:

```
if (isNST(color)) {
  return this.erc20TokenCount + this.nftTokenCount + (color - NST_COLOR_BASE);
}

if (isNFT(color)) {
  return this.erc20TokenCount + (color - NFT_COLOR_BASE);
}

return color;

```

File: src/stores/token.ts [hunk 2]
```diff
@@ -54,7 +54,7 @@ export class TokenStore extends ContractStore {
   public plasmaBalance?: BigIntType | BigIntType[];
 
   constructor(address: string, public color: number) {
-    super(isNFT(color) ? erc721 : erc20, address);
+    super(isNFT(color) || isNST(color) ? erc721 : erc20, address);
```
Comment (troggy) [inline src/stores/token.ts]: unnecessary check. `NST is NFT`, so `isNFT(color)` should suffice

File: src/stores/token.ts [hunk 3]
```diff
@@ -86,7 +86,11 @@ export class TokenStore extends ContractStore {
   }
 
   public get isNft() {
-    return isNFT(this.color);
+    return isNFT(this.color) || isNST(this.color);
```
Comment (troggy) [inline src/stores/token.ts]: ditto. unnecessary check

**Comment by troggy** [comment]
@johannbarbie I've pushed some changes, please take a look if these make sense. 

Also, do we need to show NST/DMT on wallet page? I thought these tokens are supposed to be manipulated by spending conditions only, no?

**Comment by johannbarbie** [comment]
> Also, do we need to show NST/DMT on wallet page? I thought these tokens are supposed to be manipulated by spending conditions only, no?

🤔  they are transferable, like any other token. so i would give the option to transfer them.

**Pull request merged**